src/NuGet.Core/NuGet.ProjectModel/PackageSpecExtensions.cs

```diff
@@ -8,6 +8,9 @@ namespace NuGet.ProjectModel
 {
     public static class PackageSpecExtensions
```

Comment:
I agree, extension methods should almost never be used. I'm trying to follow the current pattern, I think package spec needs some major refactoring around the way it is created/loaded, the constructors, and including helpers but that will need to happen later.